kore/src/Kore/Step/SMT/Representation/Symbols.hs

```diff
@@ -193,7 +193,11 @@ constructorDeclaration
         Attribute.Symbol.constructor attributes
     Attribute.Functional { isDeclaredFunctional } =
         Attribute.Symbol.functional attributes
-    encodedName = AST.encodable symbolConstructor
+    Attribute.Smtlib { getSmtlib } = Attribute.Symbol.smtlib attributes
+    encodedName =
```

Comment:
Just to make sure I understand correctly: This encodes the contructor's name for the solver, _unless_ the constructor symbol was given an `smtlib` attribute by the user, in which case we use the provided name. Is that right? If so, I think that's a pretty important piece of logic, and even if the function is small, I would like us to lift it to the top-level and give documentation.

Reply:
I didn't end up keeping these changes. We were debugging one of the tests and thought that this was causing an issue, but it wasn't the case.